File: SYP.Web/Modules/Setting/DailyExchanges/DailyExchangesColumns.cs
namespace SYP.Setting.Columns;

[ColumnsScript("Setting.DailyExchanges")]
[BasedOnRow(typeof(DailyExchangesRow), CheckNames = true)]
public class DailyExchangesColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public int CurrencyId { get; set; }
    [EditLink]
    public string CurrencyCode { get; set; }
    public decimal ForexBuying { get; set; }
    public decimal ForexSelling { get; set; }
    public decimal BanknoteBuying { get; set; }
    public decimal BanknoteSelling { get; set; }
    public DateTime InsertDate { get; set; }
    public int InsertUserId { get; set; }
    public int UpdateUserId { get; set; }
    public DateTime UpdateDate { get; set; }
    public bool IsActive { get; set; }
    public short DefaultExchangeTypeId { get; set; }
}

File: SYP.Web/Modules/ServerTypes/Setting/DailyExchangesColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { DailyExchangesRow } from "./DailyExchangesRow";

export interface DailyExchangesColumns {
    Id: Column<DailyExchangesRow>;
    CurrencyId: Column<DailyExchangesRow>;
    CurrencyCode: Column<DailyExchangesRow>;
    ForexBuying: Column<DailyExchangesRow>;
    ForexSelling: Column<DailyExchangesRow>;
    BanknoteBuying: Column<DailyExchangesRow>;
    BanknoteSelling: Column<DailyExchangesRow>;
    InsertDate: Column<DailyExchangesRow>;
    InsertUserId: Column<DailyExchangesRow>;
    UpdateUserId: Column<DailyExchangesRow>;
    UpdateDate: Column<DailyExchangesRow>;
    IsActive: Column<DailyExchangesRow>;
    DefaultExchangeTypeId: Column<DailyExchangesRow>;
}

export class DailyExchangesColumns extends ColumnsBase<DailyExchangesRow> {
    static readonly columnsKey = 'Setting.DailyExchanges';
    static readonly Fields = fieldsProxy<DailyExchangesColumns>();
}